### PR TITLE
Move random string generation to one place

### PR DESCRIPTION
**Pull request by henning-gerhardt**
There was two places where random strings are created in two different ways. Instead of a self written solution on one place and a library usage this change move the library usage to a central place and replace the self written solution.

### CODE REVIEW

**Comment by stweil** [comment]
Wouldn't it be better to eliminate the function `generateRandomString` completely?

**Comment by henning-gerhardt** [comment]
Creating a random string is a needed functionality in the application - why do you want to remove it?

**Comment by stweil** [comment]
Yes, it is a needed functionality. But it does not require local functions when there is a library function `RandomStringUtils.random` which does the same and can be used instead of the local functions.

**Comment by henning-gerhardt** [comment]
Using `RandomStringUtil.random`  with a secure random device needs a more complicated method call which must be replicated on all places. That is the reason why a wrapper method is needed. In the `Helper` class such "wrapper" method are placed and my change is doing this.

**Comment by solth** [comment]
@stweil do you approve this pull request with the explanation given above by @henning-gerhardt ?

File: Kitodo/src/test/java/org/kitodo/production/helper/HelperTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package org.kitodo.production.helper;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+
+class HelperTest {
```
Comment (stweil) [inline Kitodo/src/test/java/org/kitodo/production/helper/HelperTest.java:19]:
```suggestion
public class HelperTest {
```

[the remark this answers is not in the document]
Comment (stweil) [reply Kitodo/src/test/java/org/kitodo/production/helper/HelperTest.java:19]: Maybe this code scanning issue is fixed by my suggested change.

**Comment by henning-gerhardt** [comment]
> Do we really need a cryptographically strong random number generator in class `UserGenerator`? As far as I see that class is only useful for tests, so I think the code could be simplified by using simple methods from class `RandomStringUtils`.

The `RandomStringUtils` class is using a weak random generator source by default. This was the reason why this change (and an already merged pull request) was made. Using weak random generators in tests is not a good choice as tests should even use a strong / secure generator source if they are available.

**Comment by stweil** [comment]
Why should tests use strong and secure random generator sources? I think that tests are even an example where such generator sources are undesired because they use resources (entropy) although there is no need for increased security in tests. In addition pseudo random generator sources have advantages for tests because they optionally allow reproducible tests if the same seed is used.

**Comment by henning-gerhardt** [comment]
Okay, then I will close this pull request as the change is not needed or not wanted.

**Pull request closed without merging**